File: TurfPro-main/lib/models/turf/turf.dart
class Turf {
  final String id;
  final String name;
  final String imageUrl;
  final String games;
  final String location;
  final String openHours;
  final String discount;
  final double rating;

  Turf({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.games,
    required this.location,
    required this.openHours,
    required this.discount,
    required this.rating,
  });

  factory Turf.fromJson(Map<String, dynamic> json) {
    return Turf(
      id: json['_id'], 
      name: json['name'],
      imageUrl: json['imageUrl'],
      games: json['games'],
      location: json['location'],
      openHours: json['openHours'],
      discount: json['discount'],
      rating: json['rating'].toDouble(),
    );
  }
}

File: TurfPro-main/backend/routes/turf.js
const express = require('express');
const Turf = require('../models/Turf');
const multer = require('multer');
const path = require('path');

const router = express.Router();

const storage = multer.diskStorage({
    destination: (req, file, cb) => {
        cb(null, 'uploads/');
    },
    filename: (req, file, cb) => {
        cb(null, Date.now() + path.extname(file.originalname));
    }
});

const upload = multer({ storage });

router.post('/addTurf', upload.single('image'), async (req, res) => {
    const { name, games, location, openHours, discount, rating } = req.body;
    const imageUrl = req.file.path;

    try {
        const turf = new Turf({ name, imageUrl, games, location, openHours, discount, rating });
        await turf.save();
        res.status(201).json(turf);
    } catch (error) {
        res.status(400).json({ message: 'Error adding turf', error: error.message });
    }
});

router.get('/', async (req, res) => {
    try {
        const turfs = await Turf.find();
        res.status(200).json(turfs);
    } catch (error) {
        res.status(500).json({ message: 'Error fetching turfs', error: error.message });
    }
});

module.exports = router;
